mlflow/models/model.py

```diff
@@ -747,9 +763,36 @@ def log(
                     await_registration_for=await_registration_for,
                     local_model_path=local_path,
                 )
-        model_info = mlflow_model.get_model_info()
-        if registered_model is not None:
-            model_info.registered_model_version = registered_model.version
+            model_info = mlflow_model.get_model_info()
+            if registered_model is not None:
+                model_info.registered_model_version = registered_model.version
+
+            # validate input example works for serving when logging the model
+            serving_input = mlflow_model.get_serving_input(local_path)
+            if mlflow_model.signature is None and serving_input is None:
+                _logger.warning(
+                    "Input example should be provided to infer model signature if the model "
+                    "signature is not provided when logging the model."
+                )
+            if serving_input:
+                from mlflow.pyfunc import validate_serving_input
+
+                try:
+                    validate_serving_input(model_info.model_uri, serving_input)
+                except Exception as e:
+                    _logger.warning(
+                        f"Failed to validate serving input example {serving_input}. "
+                        "Alternatively, you can avoid passing input example and pass model "
+                        "signature instead when logging the model. To ensure the input example "
+                        "is valid prior serving, please try calling "
+                        "`mlflow.pyfunc.validate_serving_input` on the model uri and serving "
+                        "input example. A serving input example can be generated from model "
+                        "input example using "
+                        "`mlflow.models.utils.convert_input_example_to_serving_input` function.\n"
```

Comment:
Can we expose `mlflow.models.utils.convert_input_example_to_serving_input` in `mlflow.models`, not `mlflow.models.utils`?